core/src/main/scala/org/http4s/headers/X-B3-Flags.scala

```diff
@@ -17,12 +17,17 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import cats.syntax.all._
+import cats.parse.Parser
+import cats.parse.Numbers
 import org.http4s.util.Writer
 
 object `X-B3-Flags` extends HeaderKey.Internal[`X-B3-Flags`] with HeaderKey.Singleton {
   override def parse(s: String): ParseResult[`X-B3-Flags`] =
-    HttpHeaderParser.X_B3_FLAGS(s)
+    ParseResult.fromParser(parser, "X-B3-Flags header")(s)
+
+  private[http4s] val parser: Parser[`X-B3-Flags`] =
+    (Numbers.digits <* Parser.end).map(str => fromLong(str.toLong))
```

Comment:
I don't think this `Parser.end` is right.  It won't allow this to be composed with other parsers with remaining output.  The `parseAll` in `fromParser` should do what we want.

Reply:
It ate my earlier comment.  In response to your questoin, I think whitespace is the header's responsibility.  We're at the level of `field-value` here:

```
   header-field = field-name ":" OWS field-value OWS
```

Reply:
thanks for confirming, removed the `Parser.end`